File: backend/src/main/java/vr/balance/app/DTO/response/PatientDetailResponse.java
package vr.balance.app.DTO.response;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Data
public class PatientDetailResponse {
    UserProfileResponse user;
    List<CompletedExerciseResponse> recentExercises;
    List<CompletedExerciseResponse> recentBalanceTests;
    List<BalanceTestResponse> recentBalanceTestResults; // For the chart in the website
}

File: frontend/src/DTO/response/PatientDetailResponse.ts
import type { UserProfileResponse } from '@/DTO/response/UserProfileResponse.ts'
import type { CompletedExerciseResponse } from '@/DTO/response/CompletedExerciseResponse.ts'
import type { BalanceTestResponse } from '@/DTO/response/BalanceTestResponse.ts'

export type PatientDetailResponse = {
  user: UserProfileResponse
  recentExercises: CompletedExerciseResponse[],
  downloadLast10Exercises: CompletedExerciseResponse[],
  recentBalanceTests: BalanceTestResponse[],
}
